src/MICore/Transports/VsCodeTerminalTransport.cs

```diff
@@ -0,0 +1,305 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DebugEngineHost;
+
+namespace MICore
+{
+    public class VsCodeTerminalTransport : StreamTransport
```

Comment:
>VsCode [](start = 17, length = 6)

Up to you. But we might want to not put 'VsCode' in these names since hopefully this feature will make it to VS as well someday...